src/analysis/utils/disk-utils.ts

```diff
@@ -53,7 +53,10 @@ export const getCurrentAttachedDataDisk = (
   app: App,
   appDataDisks: PersistentDisk[]
 ): DecoratedPersistentDisk | undefined => {
-  const currentDisk: PersistentDisk | undefined = _.find({ name: app?.diskName }, appDataDisks);
+  const currentDisk: PersistentDisk | undefined =
```

Comment:
@mlilynolting lmk if this is what you had in mind

Reply:
I think this is not quite right still. `!` operator evaluates to a boolean - so `!app` and `!!app` will return `false` or `true`. So `!!app === null` will always be false (true | false <> null).

I think the unhappy paths we're guarding against in this method are

1. `app` could be `undefined` => method should return `undefined`
2. `app.diskName` could be `null` => method should return `undefined`
3. `app.diskName` could be `""`, the empty string => this is fine, continue
4. `appDataDisks` could contain no matches for `app.diskName` => return `undefined`

Reply:
Ignore this if it's old hat for you; here's how `foo?.bar` evaluates given different inital states:

<table>
<tr>
 <td>`foo = undefined`
 <td> `undefined`
<tr>
 <td> `foo = {}`
 <td> `undefined`
<tr>
 <td> `foo = { bar: undefined }`
 <td> `undefined`
<tr>
 <td> `foo = { bar: null }`
 <td> `null`
<tr>
 <td> `foo = { bar: "" }`
 <td> `""`
<tr>
 <td> `foo = { bar: "baz" }`
 <td> `"baz"`
</table> 

Here's what `foo.bar` (without the nullsafe `?.` operator) returns:

<table>
<tr>
 <td>`foo = undefined`
 <td> `Uncaught TypeError: g is undefined`
<tr>
 <td> `foo = {}`
 <td> `undefined`
<tr>
 <td> `foo = { bar: undefined }`
 <td> `undefined`
<tr>
 <td> `foo = { bar: null }`
 <td> `null`
<tr>
 <td> `foo = { bar: "" }`
 <td> `""`
<tr>
 <td> `foo = { bar: "baz" }`
 <td> `"baz"`
</table>

Reply:
but in any case this code will work as written, approved!